server/src/main/java/org/apache/lucene/search/uhighlight/CustomUnifiedHighlighter.java

```diff
@@ -141,7 +143,21 @@ public Snippet[] highlightField(LeafReader reader, int docId, CheckedSupplier<St
             return null;
         }
         int fieldValueLength = fieldValue.length();
-        if ((offsetSource == OffsetSource.ANALYSIS) && (fieldValueLength > maxAnalyzedOffset)) {
+
+        if (fieldMaxAnalyzedOffset > maxAnalyzedOffset) {
+            throw new IllegalArgumentException(
+                "max_analyzer_offset has exceeded [" + maxAnalyzedOffset + "] - maximum allowed to be analyzed for highlighting. "
```

Comment:
:+1, may be hint what to do in this case `"This maximum can be set by changing the ["
                    + IndexSettings.MAX_ANALYZED_OFFSET_SETTING.getKey()
                    + "] index level setting. "`